docs/guides/getting-started/setup/nextjs.mdx

```diff
@@ -0,0 +1,192 @@
+---
+description: Build a Tauri app using NextJS as the frontend build tool
+pagination_next: guides/architecture/README
+sidebar_custom_props:
+  doc_card_image: /img/guides/getting-started/setup/vite.svg
+---
+
+import TauriInit from './_fragments/_tauri-init.mdx'
+import Commands from './_fragments/_commands.mdx'
+import Intro from './_fragments/_intro.mdx'
+import Tabs from '@theme/Tabs'
+import TabItem from '@theme/TabItem'
+import Command from '@theme/Command'
+
+# NextJS
+
+This guide will walk you through creating your first Tauri app using the React framework [nextjs].
+
+<Intro />
+
+Here's a preview of what we will be building:
+
+![Application Preview](./next-light.png#gh-light-mode-only)
+![Application Preview](./next-dark.png#gh-dark-mode-only)
+
+## Create the Frontend
+
+[NextJS] is a React Framework that came with both `SSR` and `SSG` capability, to make it work with tauri we are only going to use SSG mode.
```

Comment:
```suggestion
[NextJS] is a React Framework that comes with both Server-Side Rendering (SSR) and Server-Side Generation (SSG) capabilities. To make NextJS work with Tauri we are going to use the SSG mode since it generates only static files that can be included in the final binary.
```

Maybe you can also explain in a short paragraph why SSG is great for app UX (fast initial rendering, fast navigation etc.) but also explain the downside of using NextJS in SSG mode (no serverless functions etc.)